STACpopulator/populator_base.py

```diff
@@ -221,3 +226,55 @@ def ingest(self) -> None:
 
             counter += 1
             LOGGER.info(f"Processed {counter} data items. {failures} failures")
+            
+
+
+class THREDDSRunner:
```

Comment:
Pending. I started working on abstracting the "runner" logic, but realized it would break the current CMIP implementation, so decided to focus only on the extension part. 

I can remove it from the PR and come back to it later.